extensions/json-jackson/src/main/java/io/deephaven/json/jackson/Parsing.java

```diff
@@ -0,0 +1,303 @@
+        return parser.getText();
+    }
+
+    static String parseBoolAsString(JsonParser parser) throws IOException {
+        return parser.getText();
+    }
+
+    static BigDecimal parseDecimalAsBigDecimal(JsonParser parser) throws IOException {
+        return parser.getDecimalValue();
+    }
+
+    static float parseNumberAsFloat(JsonParser parser) throws IOException {
+        // TODO: improve after https://github.com/FasterXML/jackson-core/issues/1229
+        return parser.getFloatValue();
+    }
+
+    static double parseNumberAsDouble(JsonParser parser) throws IOException {
+        // TODO: improve after https://github.com/FasterXML/jackson-core/issues/1229
+        return parser.getDoubleValue();
+    }
+
+    static int parseStringAsInt(JsonParser parser) throws IOException {
+        if (parser.hasTextCharacters()) {
+            // TODO: potential to write parseInt optimized for char[]
+            final int len = parser.getTextLength();
+            final CharSequence cs = CharBuffer.wrap(parser.getTextCharacters(), parser.getTextOffset(), len);
+            return Integer.parseInt(cs, 0, len, 10);
+        } else {
+            return Integer.parseInt(parser.getText());
+        }
+    }
+
+    static int parseDecimalStringAsTruncatedInt(JsonParser parser) throws IOException {
+        // parse as double then cast to int; no loss of whole number part (64 bit -> 32 bit)
+        return (int) Parsing.parseStringAsDouble(parser);
+    }
+
+    static long parseStringAsLong(JsonParser parser) throws IOException {
+        if (parser.hasTextCharacters()) {
+            // TODO: potential to write parseInt optimized for char[]
+            final int len = parser.getTextLength();
+            final CharSequence cs = CharBuffer.wrap(parser.getTextCharacters(), parser.getTextOffset(), len);
+            return Long.parseLong(cs, 0, len, 10);
+        } else {
+            return Long.parseLong(parser.getText());
+        }
+    }
+
+    static long parseDecimalStringAsLossyLong(JsonParser parser) throws IOException {
+        // parser as double then cast to long; loses info (64 bit -> 64 bit)
+        return (long) parseStringAsDouble(parser);
+    }
+
+    static long parseDecimalStringAsTruncatedLong(JsonParser parser) throws IOException {
+        // To ensure 64-bit in cases where the string is a decimal, we need BigDecimal
+        return parseStringAsBigDecimal(parser).longValue();
+    }
+
+    static long parseDecimalStringAsScaledTruncatedLong(JsonParser parser, int n) throws IOException {
+        // To ensure 64-bit in cases where the string is a decimal, we need BigDecimal
+        return parseStringAsBigDecimal(parser).scaleByPowerOfTen(n).longValue();
+    }
+
+    static float parseStringAsFloat(JsonParser parser) throws IOException {
+        // TODO: improve after https://github.com/FasterXML/jackson-core/issues/1229
+        return parser.isEnabled(StreamReadFeature.USE_FAST_DOUBLE_PARSER)
```

Comment:
What does an `isEnabled` check cost? I asked if it costs more than we save, and you said "definitely not", but that sounds like an assumption.

Reply:
```java
public boolean enabledIn(int flags) { return (flags & _mask) != 0; }
```